decompiler/pipeline/ssa/phi_lifting.py

```diff
@@ -33,6 +33,8 @@ def lift(self) -> None:
             self._lift_phi_functions_of(basic_block)
             self._remove_phi_instructions_of(basic_block)
 
+        self._cfg.root = self._cfg.get_roots()[0]
```

Comment:
But that's exactly what i want to do? 
I want to update the root, so the dominator tree gets renewed for the next use (otherwise the "new root" is missing). 

What would be  a "good way" to set the root?

Reply:
Sorry, first I thought you wrote `self._cfg._root`. How you set it is okay.
SO, yes, this works, but you also set a new root, when it does not change, and getting all roots takes linear time. So it would be more efficient to to it at the other location.

Reply:
Moved